lib/clients/authenticated.js

```diff
@@ -1,332 +1,240 @@
-  var self = this;
-
-  PublicClient.call(self, '', apiURI);
-  self.key = key;
-  self.b64secret = b64secret;
-  self.passphrase = passphrase;
-};
-
-util.inherits(AuthenticatedClient, PublicClient);
-
-_.assign(AuthenticatedClient.prototype, new function() {
-  var prototype = this;
-
-  prototype.request = function(method, uriParts, opts, callback) {
-    var self = this;
-
-    opts = opts || {};
-    if (!callback && (typeof opts === 'function')) {
+const util = require('util');
+const signRequest = require('../../lib/request_signer').signRequest;
+const request = require('request');
+const async = require('async');
+
+const PublicClient = require('./public.js');
+
+class AuthenticatedClient extends PublicClient {
+  constructor(key, b64secret, passphrase, apiURI) {
+    super('', apiURI);
+    this.key = key;
+    this.b64secret = b64secret;
+    this.passphrase = passphrase;
+  }
+
+  request(method, uriParts, opts = {}, callback) {
+    if (!callback && typeof opts == 'function') {
       callback = opts;
       opts = {};
     }
 
-    if (!callback) {
-      throw "Must supply a callback"
-    }
+    if (!callback) callback = () => {};
 
-    var relativeURI = self.makeRelativeURI(uriParts);
+    const relativeURI = this.makeRelativeURI(uriParts);
     method = method.toUpperCase();
-    _.assign(opts, {
-      'method': method,
-      'uri': self.makeAbsoluteURI(relativeURI)
+    Object.assign(opts, {
+      method: method,
+      uri: this.makeAbsoluteURI(relativeURI),
     });
 
-    self.addHeaders(opts, self._getSignature(method, relativeURI, opts));
-    request(opts, self.makeRequestCallback(callback));
-  };
+    this.addHeaders(opts, this._getSignature(method, relativeURI, opts));
+    return new Promise((resolve, reject) => {
+      request(opts, this.makeRequestCallback(callback, resolve, reject));
+    });
+  }
 
-  prototype._getSignature = function(method, relativeURI, opts) {
-    var auth = {
+  _getSignature(method, relativeURI, opts) {
+    const auth = {
       key: this.key,
       secret: this.b64secret,
-      passphrase: this.passphrase
+      passphrase: this.passphrase,
     };
-    var sig = signRequest(auth, method, relativeURI, opts);
-    if (opts.body) {
-      opts.body = JSON.stringify(opts.body);
-    }
+    const sig = signRequest(auth, method, relativeURI, opts);
+
+    if (opts.body) opts.body = JSON.stringify(opts.body);
     return {
       'CB-ACCESS-KEY': sig.key,
       'CB-ACCESS-SIGN': sig.signature,
       'CB-ACCESS-TIMESTAMP': sig.timestamp,
       'CB-ACCESS-PASSPHRASE': sig.passphrase,
     };
-  };
-
-  _.forEach(['get', 'post', 'put', 'delete'], function(method) {
-    prototype[method] = _.partial(prototype.request, method);
-  });
+  }
 
-  prototype.getAccounts = function(callback) {
-    var self = this;
-    return prototype.get.call(self, ['accounts'], callback);
-  };
+  getAccounts(callback) {
+    return this.get(['accounts'], callback);
```

Comment:
Not until [ES Class Fields](https://github.com/tc39/proposal-class-fields) are finalized (currently stage 2)